string message_id
string agent_id
string map
int32 lm_x
int32 lm_y